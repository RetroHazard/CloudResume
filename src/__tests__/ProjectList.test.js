import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ProjectList from '../components/project_list';

// Sample JSON data
const sampleData = {
    Projects: [
        {
            name: 'Generic Project',
            company: 'Generic Company',
            category: 'Generic Category',
            role: 'Generic Role',
            start: '2020',
            end: '2021',
            logo: './images/generic_logo.png',
            details: [
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
                'Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
                'Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris.',
                'Nisi ut aliquip ex ea commodo consequat.',
            ],
            links: [
                {
                    demo: [
                        {
                            website: 'https://generic-demo.com',
                            icon: 'fa-solid:external-link-alt',
                        },
                    ],
                    repo: [
                        {
                            website: 'https://generic-repo.com',
                            icon: 'simple-icons:git',
                        },
                    ],
                    spec: [
                        {
                            website: 'https://generic-spec.com',
                            icon: 'fa6-solid:info',
                        },
                    ],
                },
            ],
            technologies: [
                {
                    name: 'Generic Skill 1',
                    logo: 'ph:generic-icon-1',
                    website: 'https://generic-skill1.com',
                },
                {
                    name: 'Generic Skill 2',
                    logo: 'ph:generic-icon-2',
                    website: 'https://generic-skill2.com',
                },
                {
                    name: 'Generic Skill 3',
                    logo: 'ph:generic-icon-3',
                    website: 'https://generic-skill3.com',
                },
            ],
        },
    ],
};

// Mock the DataLoader component
jest.mock('../utils/dataLoader', () => ({
    __esModule: true,
    default: ({ children }) => children(sampleData),
}));

describe('ProjectList Component', () => {
    it('renders correctly', () => {
        const { getByText, getByAltText, getAllByRole } = render(<ProjectList />);

        expect(getByText('Generic Project')).toBeInTheDocument();
        expect(getByText('Generic Company')).toBeInTheDocument();
        expect(getByText('Generic Category')).toBeInTheDocument();
        expect(getByText('Generic Role')).toBeInTheDocument();
        expect(getByText('2020 - 2021')).toBeInTheDocument();
        expect(getByAltText('Generic Project logo')).toBeInTheDocument();

        sampleData.Projects[0].details.forEach((detail) => {
            expect(getByText(detail)).toBeInTheDocument();
        });

        const links = getAllByRole('link');
        expect(links.length).toBe(6);

        sampleData.Projects[0].technologies.forEach((skill) => {
            expect(getByText(skill.name)).toBeInTheDocument();
        });
    });

    it('matches snapshot', () => {
        const { container } = render(<ProjectList />);
        expect(container).toMatchSnapshot();
    });
});
