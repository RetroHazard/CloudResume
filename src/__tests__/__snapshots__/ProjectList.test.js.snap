// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList Component matches snapshot 1`] = `
<div>
  <div>
    <div
      class="flex flex-col gap-6"
    >
      <div
        class="flex flex-col gap-4"
      >
        <div
          class="flex w-full flex-col gap-4"
        >
          <div
            class="flex gap-4"
          >
            <img
              alt="Generic Project logo"
              class="hidden h-[7rem] w-[7rem] rounded-xl sm:block"
              src="./images/generic_logo.png"
            />
            <div
              class="flex w-full justify-between"
            >
              <div
                class="flex flex-col"
              >
                <h3
                  class="h3 mb-0 font-extrabold text-content-title max-sm:text-sm sm:text-xs md:text-base"
                >
                  Generic Project
                </h3>
                <p
                  class="mb-0.5 font-semibold leading-snug text-content-subtitle max-sm:text-sm sm:text-xs md:text-base"
                >
                  Generic Company
                </p>
                <p
                  class="mb-0 font-medium leading-snug text-content-accent max-sm:text-xs sm:text-xs md:text-sm"
                >
                  Generic Role
                </p>
                <p
                  class="mb-0 font-medium leading-snug text-content-date max-sm:text-xs sm:text-xs md:text-sm"
                >
                  2020
                   - 
                  2021
                </p>
                <p
                  class="mb-0 font-light leading-snug text-content-accent max-sm:text-xs sm:text-xs md:text-xs"
                >
                  Generic Category
                </p>
              </div>
              <div
                class="flex flex-wrap gap-3 max-sm:flex-col sm:flex-col"
              >
                <a
                  aria-label="demo"
                  class="social-link"
                  href="https://generic-demo.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <iconify-icon
                    height="1.25em"
                    icon="fa-solid:external-link-alt"
                    width="1.25em"
                  />
                </a>
                <a
                  aria-label="repo"
                  class="social-link"
                  href="https://generic-repo.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <iconify-icon
                    height="1.25em"
                    icon="simple-icons:git"
                    width="1.25em"
                  />
                </a>
                <a
                  aria-label="spec"
                  class="social-link"
                  href="https://generic-spec.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <iconify-icon
                    height="1.25em"
                    icon="fa6-solid:info"
                    width="1.25em"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mb-3 font-normal leading-relaxed max-sm:text-xs sm:text-xs sm:leading-relaxed md:text-sm"
        >
          <ul
            class="list-disc pl-10 pr-5"
          >
            <li>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </li>
            <li>
              Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            </li>
            <li>
              Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris.
            </li>
            <li>
              Nisi ut aliquip ex ea commodo consequat.
            </li>
          </ul>
        </div>
      </div>
      <div
        class="flex flex-col gap-1.5"
      >
        <span
          class="font-medium leading-snug text-content-accent max-sm:text-xs sm:text-xs md:text-sm"
        >
          Technologies:
        </span>
        <div
          class="flex flex-wrap max-sm:gap-1 sm:gap-2 md:gap-3"
        >
          <a
            class="skill-block"
            href="https://generic-skill1.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <iconify-icon
              class="icon-box"
              classname="icon-box"
              height="1.1em"
              icon="ph:generic-icon-1"
              width="1.1em"
            />
            Generic Skill 1
          </a>
          <a
            class="skill-block"
            href="https://generic-skill2.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <iconify-icon
              class="icon-box"
              classname="icon-box"
              height="1.1em"
              icon="ph:generic-icon-2"
              width="1.1em"
            />
            Generic Skill 2
          </a>
          <a
            class="skill-block"
            href="https://generic-skill3.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <iconify-icon
              class="icon-box"
              classname="icon-box"
              height="1.1em"
              icon="ph:generic-icon-3"
              width="1.1em"
            />
            Generic Skill 3
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
